### ISSUE TITLE
Destroy enemy walls: Siege engine

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1184]
A nice addition to the game would be some way to breach the walls of a fully walled in player. This possibility does not need to be there on every map, but it being there will certainly provide interesting situations.

This method of breaking walls should NOT be a simple - even if expensive - 'destroy wall' spell for two very important reasons:
1. Heroes without a dungeon do not use spells and a mapmaker would want to create parties that can breach walls.
2. The mechanics behind a destroy wall spell are boring for both the caster and the player on the receiving end. More on this topic see [here.](http://forum.freegamedev.net/viewtopic.php?f=37&t=6858) Simply put if a player has lots of gems then enemy walls become meaningless and there's no defense against it.

Instead I propose to add a battering ram or some other type of siege engine that is wheeled to the dungeon wall, has to be deployed and slowly starts to break down the wall alerting the player who is under attack.
This gives the player on the receiving end time to respond in either building defenses at the breach point or by taking the offensive and possibly destroy the siege engine.

A mapmaker should be able to spawn heroes with a siege weapon if he wants to breach the player walls. Players should be able to somehow construct this, possibly in the workshop, when the map is configured to allow it.

For inspiration:
![n5khn6](https://cloud.githubusercontent.com/assets/13840686/14943545/70235340-0fdc-11e6-9d18-e5696826bc53.jpg)

**Comment by hwoarangmy** [comment]
I like your idea of having something not as simple as a spell that would unclaim the wall.

**Comment by Loobinex** [comment]
Thanks, some additional strategy is always nice.

I would suggest to have the wall destroyed though, not unclaimed.

**Comment by hwoarangmy** [comment]
yes, I can see no reason to want to unclaim an enemy wall without destroying it

**Comment by hwoarangmy** [comment]
I've thought about something that may be interesting here. We could add a new room working like the workshop (ie producing some gems) that would be required for some kind of spells (very powerfull). Your siege weapon would be one of those. It would require gems during its powering up (and while active) and would collapse if none is brought for too long.
The spell would be so powerfull that every player see that it is cast (so that a player knows there is a risk here - but we may just display some generic spell animation so that a player do not exactly know which kind of spell has been cast).

@OpenDungeons/developers WDYT ?

**Comment by hwoarangmy** [comment]
And for heroes, we could just let the portal spawn the gems with workers and cast the spell

**Comment by Loobinex** [comment]
I'm not entirely sure what you've got in mind, but what is important for the destroy wall mechanism is that the receiving player can know which wall is being attacked and has time to react to this very location, and has a possibility to stop it directly - but only by opening up another section of wall.

**Comment by hwoarangmy** [comment]
> I'm not entirely sure what you've got in mind

ATM, when you build a trap, it is not active once you create it. It has to be crafted in the workshop and brought by a worker. The idea here would be to do something similar (in a dedicated room). That would allow to have the time for the attacked player to do something if he is warned once the spell is cast while the "spell gems" are being created and brought. Moreover, it would add some strategy since it will be faster if the room is not far from the casted spell (and workers are around).

**Comment by Loobinex** [comment]
Being alerted that a spell is being cast is not as nice as hearing that your walls are in the process of being assaulted. If the siege engine actually has to be placed against the wall it is trying to breach for a while you can attack there. If the timing takes place in the player 'workshop' there's not as much you can do.
Of course you could still have it - or separate components - placed the same way traps are now.

Important to note is that building and defending dungeons (with walls/traps) is a key element of this game, and if you're going for a multiplayer focus then the 'breaking down of walls' game play will be one of the most important aspects of the game, and should not be merely an afterthought. It might be worth it to do some prototyping to see what results in the most fun games in both offense and defense.

The Dungeon Keeper games did have no such thing as - especially the first one - were focused on single player.
